[BUG] JobManager refactoring creates a bad behaviour with `filterJobStateTransition`

`JobManager` if a client calls `killJob`, DIRAC itself will call a client, with DIRAC "auth power".

# Why

1. At the beginning, `killJob` is in an handler with DIRAC credentials and not the user's. 
2. After #8244 's refactoring, `filterJobStateTransition` calls `res = JobMonitoringClient().getJobsStatus(jobIDs)` which is bad because it is with DIRAC credentials, and not the user's. 

# Impact

1. In DIRAC, "it works" as DIRAC has its own credentials, but it's not an expected behaviour.
2. With DiracX, it maybe won't work as DIRAC is not a valid user in the CS.
3. In CI with DiracX (how it was detected), it won't work because DIRAC own certificate used for DiracX does not exist:

```bash
AssertionError: FileNotFoundError(2, 'No such file or directory'): [Errno 2] No such file or directory: '/tmp/x509up_u501'

# Stack
68.117.5]:57294)[prod:ciuser] (0.13 secs) ERROR: FileNotFoundError(2, 'No such file or directory'): [Errno 2] No such file or directory: '/tmp/x509up_u501'
  File "/home/dirac/LocalRepo/ALTERNATIVE_MODULES/DIRAC/src/DIRAC/Core/Utilities/ReturnValues.py", line 238, in wrapped
    value = func(*args, **kwargs)
            ^^^^^^^^^^^^^^^^^^^^^
  File "/home/dirac/LocalRepo/ALTERNATIVE_MODULES/DIRAC/src/DIRAC/WorkloadManagementSystem/FutureClient/JobMonitoringClient.py", line 35, in getJobsStatus
    return self.fetch(["Status"], jobIDs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/dirac/LocalRepo/ALTERNATIVE_MODULES/DIRAC/src/DIRAC/WorkloadManagementSystem/FutureClient/JobMonitoringClient.py", line 10, in fetch
    with DiracXClient() as api:
  File "/home/dirac/ServerInstallDIR/diracos/lib/python3.11/contextlib.py", line 137, in __enter__
    return next(self.gen)
           ^^^^^^^^^^^^^^
  File "/home/dirac/LocalRepo/ALTERNATIVE_MODULES/DIRAC/src/DIRAC/Core/Security/DiracX.py", line 99, in DiracXClient
    diracxToken = diracxTokenFromPEM(proxyLocation)
                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/dirac/LocalRepo/ALTERNATIVE_MODULES/DIRAC/src/DIRAC/Core/Security/DiracX.py", line 77, in diracxTokenFromPEM
    pem = Path(pemPath).read_text()
          ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/dirac/ServerInstallDIR/diracos/lib/python3.11/pathlib.py", line 1058, in read_text
    with self.open(mode='r', encoding=encoding, errors=errors) as f:
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/dirac/ServerInstallDIR/diracos/lib/python3.11/pathlib.py", line 1044, in open
    return io.open(self, mode, buffering, encoding, errors, newline)
```

> Note: Maybe there's other weird calls like this... We only detected it because of the CI with DiracX

--- a/src/DIRAC/Interfaces/API/Dirac.py
+++ b/src/DIRAC/Interfaces/API/Dirac.py
@@ -46,6 +46,7 @@ from DIRAC.WorkloadManagementSystem.Client import JobStatus
 from DIRAC.WorkloadManagementSystem.Client.JobMonitoringClient import JobMonitoringClient
 from DIRAC.WorkloadManagementSystem.Client.SandboxStoreClient import SandboxStoreClient
 from DIRAC.WorkloadManagementSystem.Client.WMSClient import WMSClient
+from DIRAC.WorkloadManagementSystem.Utilities.jobAdministration import _filterJobStateTransition
 
 
 def parseArguments(args):
@@ -1450,10 +1451,13 @@ class Dirac(API):
         # Remove any job IDs that can't change to the Killed or Deleted states
         filteredJobs = set()
         for filterState in (JobStatus.KILLED, JobStatus.DELETED):
-            filterRes = JobStatus.filterJobStateTransition(jobIDs, filterState)
-            if not filterRes["OK"]:
-                return filterRes
-            filteredJobs.update(filterRes["Value"])
+            # get a dictionary of jobID:status
+            res = JobMonitoringClient().getJobsStatus(jobIDs)
+            if not res["OK"]:
+                return res
+            js = {k: v["Status"] for k, v in res["Value"].items()}
+            # then filter
+            filteredJobs.update(_filterJobStateTransition(js, filterState))
 
         return WMSClient(useCertificates=self.useCertificates).deleteJob(list(filteredJobs))
 
@@ -1480,11 +1484,13 @@ class Dirac(API):
             return ret
         jobIDs = ret["Value"]
 
-        # Remove any job IDs that can't change to the rescheduled state
-        filterRes = JobStatus.filterJobStateTransition(jobIDs, JobStatus.RESCHEDULED)
-        if not filterRes["OK"]:
-            return filterRes
-        jobIDsToReschedule = filterRes["Value"]
+        # get a dictionary of jobID:status
+        res = JobMonitoringClient().getJobsStatus(jobIDs)
+        if not res["OK"]:
+            return res
+        js = {k: v["Status"] for k, v in res["Value"].items()}
+        # then filter
+        jobIDsToReschedule = _filterJobStateTransition(js, JobStatus.RESCHEDULED)
 
         return WMSClient(useCertificates=self.useCertificates).rescheduleJob(jobIDsToReschedule)
 
@@ -1510,10 +1516,13 @@ class Dirac(API):
         # Remove any job IDs that can't change to the Killed or Deleted states
         filteredJobs = set()
         for filterState in (JobStatus.KILLED, JobStatus.DELETED):
-            filterRes = JobStatus.filterJobStateTransition(jobIDs, filterState)
-            if not filterRes["OK"]:
-                return filterRes
-            filteredJobs.update(filterRes["Value"])
+            # get a dictionary of jobID:status
+            res = JobMonitoringClient().getJobsStatus(jobIDs)
+            if not res["OK"]:
+                return res
+            js = {k: v["Status"] for k, v in res["Value"].items()}
+            # then filter
+            filteredJobs.update(_filterJobStateTransition(js, filterState))
 
         return WMSClient(useCertificates=self.useCertificates).killJob(list(filteredJobs))
 

--- a/src/DIRAC/WorkloadManagementSystem/Agent/test/Test_Agent_StalledJobAgent.py
+++ b/src/DIRAC/WorkloadManagementSystem/Agent/test/Test_Agent_StalledJobAgent.py
@@ -28,6 +28,7 @@ def sja(mocker):
     mocker.patch("DIRAC.WorkloadManagementSystem.Agent.StalledJobAgent.rescheduleJobs", return_value=MagicMock())
     mocker.patch("DIRAC.WorkloadManagementSystem.Agent.StalledJobAgent.PilotAgentsDB", return_value=MagicMock())
     mocker.patch("DIRAC.WorkloadManagementSystem.Agent.StalledJobAgent.getJobParameters", return_value=MagicMock())
+    mocker.patch("DIRAC.WorkloadManagementSystem.Agent.StalledJobAgent.kill_delete_jobs", return_value=MagicMock())
 
     stalledJobAgent = StalledJobAgent()
     stalledJobAgent._AgentModule__configDefaults = mockAM

--- a/src/DIRAC/WorkloadManagementSystem/Client/JobStatus.py
+++ b/src/DIRAC/WorkloadManagementSystem/Client/JobStatus.py
@@ -2,9 +2,7 @@
 This module contains constants and lists for the possible job states.
 """
 
-from DIRAC import S_OK
 from DIRAC.Core.Utilities.StateMachine import State, StateMachine
-from DIRAC.WorkloadManagementSystem.Client.JobMonitoringClient import JobMonitoringClient
 
 #:
 SUBMITTING = "Submitting"
@@ -95,26 +93,3 @@ class JobsStateMachine(StateMachine):
             RECEIVED: State(1, [SCOUTING, CHECKING, STAGING, WAITING, FAILED, DELETED, KILLED], defState=RECEIVED),
             SUBMITTING: State(0, [RECEIVED, CHECKING, DELETED, KILLED], defState=SUBMITTING),  # initial state
         }
-
-
-def filterJobStateTransition(jobIDs, candidateState):
-    """Given a list of jobIDs, return a list that are allowed to transition
-    to the given candidate state.
-    """
-    allowedJobs = []
-
-    if not isinstance(jobIDs, list):
-        jobIDs = [jobIDs]
-
-    res = JobMonitoringClient().getJobsStatus(jobIDs)
-    if not res["OK"]:
-        return res
-
-    for jobID in jobIDs:
-        if jobID in res["Value"]:
-            curState = res["Value"][jobID]["Status"]
-            stateRes = JobsStateMachine(curState).getNextState(candidateState)
-            if stateRes["OK"]:
-                if stateRes["Value"] == candidateState:
-                    allowedJobs.append(jobID)
-    return S_OK(allowedJobs)

--- a/src/DIRAC/WorkloadManagementSystem/Client/JobStatus.py
+++ b/src/DIRAC/WorkloadManagementSystem/Client/JobStatus.py
@@ -2,12 +2,9 @@
 This module contains constants and lists for the possible job states.
 """
 
-from DIRAC import gLogger, S_OK, S_ERROR
+from DIRAC import S_OK
 from DIRAC.Core.Utilities.StateMachine import State, StateMachine
-from DIRAC.Core.Utilities.Decorators import deprecated
-
 from DIRAC.WorkloadManagementSystem.Client.JobMonitoringClient import JobMonitoringClient
-
 
 #:
 SUBMITTING = "Submitting"
@@ -98,37 +95,6 @@ class JobsStateMachine(StateMachine):
             RECEIVED: State(1, [SCOUTING, CHECKING, STAGING, WAITING, FAILED, DELETED, KILLED], defState=RECEIVED),
             SUBMITTING: State(0, [RECEIVED, CHECKING, DELETED, KILLED], defState=SUBMITTING),  # initial state
         }
-
-
-@deprecated("Use filterJobStateTransition instead")
-def checkJobStateTransition(jobID, candidateState, currentStatus=None, jobMonitoringClient=None):
-    """Utility to check if a job state transition is allowed"""
-    if not currentStatus:
-        if not jobMonitoringClient:
-            from DIRAC.WorkloadManagementSystem.Client.JobMonitoringClient import JobMonitoringClient
-
-            jobMonitoringClient = JobMonitoringClient()
-
-        res = jobMonitoringClient.getJobsStatus(jobID)
-        if not res["OK"]:
-            return res
-        try:
-            currentStatus = res["Value"][jobID]["Status"]
-        except KeyError:
-            return S_ERROR("Job does not exist")
-
-    res = JobsStateMachine(currentStatus).getNextState(candidateState)
-    if not res["OK"]:
-        return res
-
-    # If the JobsStateMachine does not accept the candidate, return an ERROR
-    if candidateState != res["Value"]:
-        gLogger.error(
-            "Job Status Error",
-            f"{jobID} can't move from {currentStatus} to {candidateState}",
-        )
-        return S_ERROR("Job state transition not allowed")
-    return S_OK()
 
 
 def filterJobStateTransition(jobIDs, candidateState):

--- a/src/DIRAC/WorkloadManagementSystem/FutureClient/JobStateUpdateClient.py
+++ b/src/DIRAC/WorkloadManagementSystem/FutureClient/JobStateUpdateClient.py
@@ -85,12 +85,6 @@ class JobStateUpdateClient(FutureClient):
 
     @stripValueIfOK
     @convertToReturnValue
-    def setJobFlag(self, jobID: str | int, flag: str):
-        with DiracXClient() as api:
-            api.jobs.patch_metadata({jobID: {flag: True}})
-
-    @stripValueIfOK
-    @convertToReturnValue
     def setJobParameter(self, jobID: str | int, name: str, value: str):
         with DiracXClient() as api:
             api.jobs.patch_metadata({jobID: {name: value}})
@@ -150,12 +144,6 @@ class JobStateUpdateClient(FutureClient):
         with DiracXClient() as api:
             updates = {job_id: {k: v} for job_id, (k, v) in jobsParameterDict.items()}
             api.jobs.patch_metadata(updates)
-
-    @stripValueIfOK
-    @convertToReturnValue
-    def unsetJobFlag(self, jobID: str | int, flag: str):
-        with DiracXClient() as api:
-            api.jobs.patch_metadata({jobID: {flag: False}})
 
     @stripValueIfOK
     @convertToReturnValue

--- a/src/DIRAC/WorkloadManagementSystem/Service/JobStateUpdateHandler.py
+++ b/src/DIRAC/WorkloadManagementSystem/Service/JobStateUpdateHandler.py
@@ -129,22 +129,6 @@ class JobStateUpdateHandlerMixin:
         return cls.jobDB.setJobAttribute(int(jobID), "Site", site)
 
     ###########################################################################
-    types_setJobFlag = [[str, int], str]
-
-    @classmethod
-    def export_setJobFlag(cls, jobID, flag):
-        """Set job flag for job with jobID"""
-        return cls.jobDB.setJobAttribute(int(jobID), flag, "True")
-
-    ###########################################################################
-    types_unsetJobFlag = [[str, int], str]
-
-    @classmethod
-    def export_unsetJobFlag(cls, jobID, flag):
-        """Unset job flag for job with jobID"""
-        return cls.jobDB.setJobAttribute(int(jobID), flag, "False")
-
-    ###########################################################################
     types_setJobApplicationStatus = [[str, int], str, str]
 
     @classmethod

--- a/src/DIRAC/WorkloadManagementSystem/Utilities/jobAdministration.py
+++ b/src/DIRAC/WorkloadManagementSystem/Utilities/jobAdministration.py
@@ -84,17 +84,11 @@ def kill_delete_jobs(right, validJobList, nonauthJobList=[], force=False):
         jobStates = result["Value"]
 
         # Get the jobs allowed to transition to the Killed state
-        filterRes = _filterJobStateTransition(jobStates, JobStatus.KILLED)
-        if not filterRes["OK"]:
-            return filterRes
-        killJobList.extend(filterRes["Value"])
+        killJobList.extend(_filterJobStateTransition(jobStates, JobStatus.KILLED))
 
         if right == RIGHT_DELETE:
             # Get the jobs allowed to transition to the Deleted state
-            filterRes = _filterJobStateTransition(jobStates, JobStatus.DELETED)
-            if not filterRes["OK"]:
-                return filterRes
-            deleteJobList.extend(filterRes["Value"])
+            deleteJobList.extend(_filterJobStateTransition(jobStates, JobStatus.DELETED))
 
         for jobID in killJobList:
             result = _killJob(jobID, force=force)
@@ -141,4 +135,4 @@ def _filterJobStateTransition(jobStates, candidateState):
         if stateRes["OK"]:
             if stateRes["Value"] == candidateState:
                 allowedJobs.append(js[0])
-    return S_OK(allowedJobs)
+    return allowedJobs

--- a/src/DIRAC/WorkloadManagementSystem/Utilities/jobAdministration.py
+++ b/src/DIRAC/WorkloadManagementSystem/Utilities/jobAdministration.py
@@ -1,11 +1,10 @@
 from DIRAC import S_ERROR, S_OK, gLogger
 from DIRAC.StorageManagementSystem.DB.StorageManagementDB import StorageManagementDB
 from DIRAC.WorkloadManagementSystem.Client import JobStatus
-from DIRAC.WorkloadManagementSystem.Client.JobStatus import filterJobStateTransition
 from DIRAC.WorkloadManagementSystem.DB.JobDB import JobDB
 from DIRAC.WorkloadManagementSystem.DB.PilotAgentsDB import PilotAgentsDB
 from DIRAC.WorkloadManagementSystem.DB.TaskQueueDB import TaskQueueDB
-from DIRAC.WorkloadManagementSystem.Service.JobPolicy import RIGHT_KILL, RIGHT_DELETE
+from DIRAC.WorkloadManagementSystem.Service.JobPolicy import RIGHT_DELETE, RIGHT_KILL
 
 
 def _deleteJob(jobID, force=False):
@@ -79,15 +78,20 @@ def kill_delete_jobs(right, validJobList, nonauthJobList=[], force=False):
     killJobList = []
     deleteJobList = []
     if validJobList:
+        result = JobDB().getJobsAttributes(killJobList, ["Status"])
+        if not result["OK"]:
+            return result
+        jobStates = result["Value"]
+
         # Get the jobs allowed to transition to the Killed state
-        filterRes = filterJobStateTransition(validJobList, JobStatus.KILLED)
+        filterRes = _filterJobStateTransition(jobStates, JobStatus.KILLED)
         if not filterRes["OK"]:
             return filterRes
         killJobList.extend(filterRes["Value"])
 
         if right == RIGHT_DELETE:
             # Get the jobs allowed to transition to the Deleted state
-            filterRes = filterJobStateTransition(validJobList, JobStatus.DELETED)
+            filterRes = _filterJobStateTransition(jobStates, JobStatus.DELETED)
             if not filterRes["OK"]:
                 return filterRes
             deleteJobList.extend(filterRes["Value"])
@@ -103,10 +107,7 @@ def kill_delete_jobs(right, validJobList, nonauthJobList=[], force=False):
                 badIDs.append(jobID)
 
         # Look for jobs that are in the Staging state to send kill signal to the stager
-        result = JobDB().getJobsAttributes(killJobList, ["Status"])
-        if not result["OK"]:
-            return result
-        stagingJobList = [jobID for jobID, sDict in result["Value"].items() if sDict["Status"] == JobStatus.STAGING]
+        stagingJobList = [jobID for jobID, sDict in jobStates.items() if sDict["Status"] == JobStatus.STAGING]
 
         if stagingJobList:
             stagerDB = StorageManagementDB()
@@ -127,3 +128,17 @@ def kill_delete_jobs(right, validJobList, nonauthJobList=[], force=False):
 
     jobsList = killJobList if right == RIGHT_KILL else deleteJobList
     return S_OK(jobsList)
+
+
+def _filterJobStateTransition(jobStates, candidateState):
+    """Given a dictionary of jobs states,
+    return a list of jobs that are allowed to transition to the given candidate state.
+    """
+    allowedJobs = []
+
+    for js in jobStates.items():
+        stateRes = JobStatus.JobsStateMachine(js[1]["Status"]).getNextState(candidateState)
+        if stateRes["OK"]:
+            if stateRes["Value"] == candidateState:
+                allowedJobs.append(js[0])
+    return S_OK(allowedJobs)

--- a/src/DIRAC/WorkloadManagementSystem/Utilities/test/Test_JobAdministration.py
+++ b/src/DIRAC/WorkloadManagementSystem/Utilities/test/Test_JobAdministration.py
@@ -10,31 +10,19 @@ from DIRAC.WorkloadManagementSystem.Utilities.jobAdministration import kill_dele
 
 
 @pytest.mark.parametrize(
-    "jobIDs_list, right, filtered_jobs, expected_res, expected_value",
+    "jobIDs_list, right",
     [
-        ([], "Kill", [], True, []),
-        ([], "Delete", [], True, []),
-        (1, "Kill", [], True, []),
-        (1, "Kill", [1], True, [1]),
-        ([1, 2], "Kill", [], True, []),
-        ([1, 2], "Kill", [1], True, [1]),
-        (1, "Kill", [1], True, [1]),
-        ([1, 2], "Kill", [1], True, [1]),
-        ([1, 2], "Kill", [2], True, [2]),
-        ([1, 2], "Kill", [], True, []),
-        ([1, 2], "Kill", [1, 2], True, [1, 2]),
+        ([], "Kill"),
+        ([], "Delete"),
+        (1, "Kill"),
+        ([1, 2], "Kill"),
     ],
 )
-def test___kill_delete_jobs(mocker, jobIDs_list, right, filtered_jobs, expected_res, expected_value):
+def test___kill_delete_jobs(mocker, jobIDs_list, right):
     mocker.patch("DIRAC.WorkloadManagementSystem.Utilities.jobAdministration.JobDB", MagicMock())
     mocker.patch("DIRAC.WorkloadManagementSystem.Utilities.jobAdministration.TaskQueueDB", MagicMock())
     mocker.patch("DIRAC.WorkloadManagementSystem.Utilities.jobAdministration.PilotAgentsDB", MagicMock())
     mocker.patch("DIRAC.WorkloadManagementSystem.Utilities.jobAdministration.StorageManagementDB", MagicMock())
-    mocker.patch(
-        "DIRAC.WorkloadManagementSystem.Utilities.jobAdministration.filterJobStateTransition",
-        return_value={"OK": True, "Value": filtered_jobs},
-    )
 
     res = kill_delete_jobs(right, jobIDs_list)
-    assert res["OK"] == expected_res
-    assert res["Value"] == expected_value
+    assert res["OK"]

--- a/tests/Integration/FutureClient/WorkloadManagement/Test_JobStateUpdate.py
+++ b/tests/Integration/FutureClient/WorkloadManagement/Test_JobStateUpdate.py
@@ -142,42 +142,6 @@ def test_setJobAttribute(monkeypatch, example_jobids, args):
     assert result[example_jobids[0]] == result[example_jobids[1]]
 
 
-def test_setJobFlag(monkeypatch, example_jobids):
-    # JobStateUpdateClient().setJobFlag(jobID: str | int, flag: str)
-    assert returnValueOrRaise(JobMonitoringClient().getJobAttribute(example_jobids[0], "AccountedFlag")) == "False"
-    assert returnValueOrRaise(JobMonitoringClient().getJobAttribute(example_jobids[1], "AccountedFlag")) == "False"
-
-    method = JobStateUpdateClient().setJobFlag
-    test_func1 = partial(method, example_jobids[0], "AccountedFlag")
-    test_func2 = partial(method, example_jobids[1], "AccountedFlag")
-    compare_results2(monkeypatch, test_func1, test_func2)
-
-    assert returnValueOrRaise(JobMonitoringClient().getJobAttribute(example_jobids[0], "AccountedFlag")) == "True"
-    assert returnValueOrRaise(JobMonitoringClient().getJobAttribute(example_jobids[1], "AccountedFlag")) == "True"
-
-
-def test_unsetJobFlag(monkeypatch, example_jobids):
-    # JobStateUpdateClient().unsetJobFlag(jobID: str | int, flag: str)
-    assert returnValueOrRaise(JobMonitoringClient().getJobAttribute(example_jobids[0], "AccountedFlag")) == "False"
-    assert returnValueOrRaise(JobMonitoringClient().getJobAttribute(example_jobids[1], "AccountedFlag")) == "False"
-
-    method = JobStateUpdateClient().setJobFlag
-    test_func1 = partial(method, example_jobids[0], "AccountedFlag")
-    test_func2 = partial(method, example_jobids[1], "AccountedFlag")
-    compare_results2(monkeypatch, test_func1, test_func2)
-
-    assert returnValueOrRaise(JobMonitoringClient().getJobAttribute(example_jobids[0], "AccountedFlag")) == "True"
-    assert returnValueOrRaise(JobMonitoringClient().getJobAttribute(example_jobids[1], "AccountedFlag")) == "True"
-
-    method = JobStateUpdateClient().unsetJobFlag
-    test_func1 = partial(method, example_jobids[0], "AccountedFlag")
-    test_func2 = partial(method, example_jobids[1], "AccountedFlag")
-    compare_results2(monkeypatch, test_func1, test_func2)
-
-    assert returnValueOrRaise(JobMonitoringClient().getJobAttribute(example_jobids[0], "AccountedFlag")) == "False"
-    assert returnValueOrRaise(JobMonitoringClient().getJobAttribute(example_jobids[1], "AccountedFlag")) == "False"
-
-
 @pytest.mark.parametrize(
     "args",
     [

--- a/tests/Integration/WorkloadManagementSystem/Test_Client_WMS.py
+++ b/tests/Integration/WorkloadManagementSystem/Test_Client_WMS.py
@@ -289,7 +289,9 @@ def test_JobStateUpdateAndJobMonitoring() -> None:
         # forcing the update
         res = jobStateUpdateClient.setJobStatus(jobID, JobStatus.RUNNING, "running", "source", None, True)
         assert res["OK"], res["Message"]
-        res = jobStateUpdateClient.setJobParameters(jobID, [("par1", "par1Value"), ("par2", "par2Value")])
+        res = jobStateUpdateClient.setJobParameters(jobID, [("CPUNormalizationFactor", 10), ("HostName", "hlt123")])
+        assert res["OK"], res["Message"]
+        res = jobStateUpdateClient.setJobParameter(jobID, "Pilot_Reference", "https://pilot.reference.com/123456")
         assert res["OK"], res["Message"]
         res = jobStateUpdateClient.setJobApplicationStatus(jobID, "app status", "source")
         assert res["OK"], res["Message"]
@@ -303,25 +305,25 @@ def test_JobStateUpdateAndJobMonitoring() -> None:
 
         time.sleep(3)
 
-        res = jobMonitoringClient.getJobParameters(jobID, ["par1", "par2"])
+        res = jobMonitoringClient.getJobParameters(jobID, ["CPUNormalizationFactor", "HostName"])
         assert res["OK"], res["Message"]
-        assert res["Value"] == {jobID: {"par1": "par1Value", "par2": "par2Value"}}
+        assert res["Value"] == {jobID: {"CPUNormalizationFactor": 10, "HostName": "hlt123"}}
 
-        res = jobMonitoringClient.getJobParameter(jobID, "par1")
+        res = jobMonitoringClient.getJobParameter(jobID, "Pilot_Reference")
         assert res["OK"], res["Message"]
-        assert res["Value"] == {"par1": "par1Value"}
+        assert res["Value"] == {"Pilot_Reference": "https://pilot.reference.com/123456"}
 
         res = jobMonitoringClient.getJobParameters(jobID)
         assert res["OK"], res["Message"]
         assert jobID in res["Value"]
-        assert "par1" in res["Value"][jobID]
-        assert "par2" in res["Value"][jobID]
-        assert res["Value"][jobID]["par1"] == "par1Value"
-        assert res["Value"][jobID]["par2"] == "par2Value"
+        assert "CPUNormalizationFactor" in res["Value"][jobID]
+        assert "HostName" in res["Value"][jobID]
+        assert res["Value"][jobID]["CPUNormalizationFactor"] == 10
+        assert res["Value"][jobID]["HostName"] == "hlt123"
 
-        res = jobMonitoringClient.getJobParameters(jobID, "par1")
+        res = jobMonitoringClient.getJobParameters(jobID, "Pilot_Reference")
         assert res["OK"], res["Message"]
-        assert res["Value"] == {jobID: {"par1": "par1Value"}}
+        assert res["Value"] == {jobID: {"Pilot_Reference": "https://pilot.reference.com/123456"}}
 
         res = jobMonitoringClient.getJobAttribute(jobID, "Site")
         assert res["OK"], res["Message"]
@@ -332,9 +334,12 @@ def test_JobStateUpdateAndJobMonitoring() -> None:
         assert res["Value"]["ApplicationStatus"] == "app status"
         assert res["Value"]["JobName"] == "helloWorld"
 
+        res = jobStateUpdateClient.setJobAttribute(jobID, "ApplicationStatus", "some_app_status")
+        assert res["OK"], res["Message"]
+
         res = jobMonitoringClient.getJobSummary(jobID)
         assert res["OK"], res["Message"]
-        assert res["Value"]["ApplicationStatus"] == "app status"
+        assert res["Value"]["ApplicationStatus"] == "some_app_status"
         assert res["Value"]["Status"] == JobStatus.RUNNING
 
         res = jobMonitoringClient.getJobHeartBeatData(jobID)
@@ -355,7 +360,7 @@ def test_JobStateUpdateAndJobMonitoring() -> None:
         assert res["OK"], res["Message"]
         assert res["Value"]["Status"] == JobStatus.DONE
         assert res["Value"]["MinorStatus"] == "MinorStatus"
-        assert res["Value"]["ApplicationStatus"] == "app status"
+        assert res["Value"]["ApplicationStatus"] == "some_app_status"
 
         res = jobStateUpdateClient.sendHeartBeat(jobID, {"bih": "bih"}, {"boh": "boh"})
         assert res["OK"], res["Message"]

--- a/tests/Integration/WorkloadManagementSystem/Test_Client_WMS.py
+++ b/tests/Integration/WorkloadManagementSystem/Test_Client_WMS.py
@@ -293,10 +293,6 @@ def test_JobStateUpdateAndJobMonitoring() -> None:
         assert res["OK"], res["Message"]
         res = jobStateUpdateClient.setJobApplicationStatus(jobID, "app status", "source")
         assert res["OK"], res["Message"]
-        # res = jobStateUpdate.setJobFlag()
-        # self.assertTrue(res['OK'], res.get('Message'))
-        # res = jobStateUpdate.unsetJobFlag()
-        # self.assertTrue(res['OK'], res.get('Message'))
         res = jobStateUpdateClient.setJobSite(jobID, "Site")
         assert res["OK"], res["Message"]
 

